purchase_post_code_propose_address_sst/models/purchase_order.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import json
+import logging
+import urllib
+
+from odoo import _, api, fields, models
+
+try:
+    import jaconv
+except (ImportError, IOError) as err:
+    logging.getLogger(__name__).warning(err)
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    country_id = fields.Many2one(
+        "res.country", string="Country", default=lambda self: self.env.ref("base.jp")
+    )
+    state_id = fields.Many2one("res.country.state", string="Prefecture")
+    city = fields.Char()
+    street = fields.Char()
+    street2 = fields.Char()
+    zipcode = fields.Char(string="Post Code (Search)")
```

Comment:
@kanda999 @AungKoKoLin1997 Can you consider splitting this field and a few methods (`_onchange_zipcode()` and `check_zipcode()`) into a mixin model in a separate module, and then inherit that model in the purchase order class?  The new module with the mixin model can include the extension to res.partner (let this model inherit the mixin) and be put in the Japan localization repo.  What do you think?

Reply:
I think it is a good idea and I will follow up.